File: agentic-systems/talent-lens/frontend/src/types/assessment.ts
export type ConfidenceLevel = "demonstrated" | "mentioned" | "claimed";
export type AssessmentStatus = "assessed_positive" | "assessed_negative" | "not_assessed";

export interface Evidence {
  quote: string;
  speaker: string | null;
  relevance: string | null;
}

export interface CriterionScore {
  criterion_name: string;
  score: number;
  max_score: number;
  confidence_level?: ConfidenceLevel;
  assessment_status?: AssessmentStatus;
  reasoning: string | null;
  evidence: Evidence[];
}

export interface Assessment {
  id: string;
  candidate_id: string;
  interview_id: string | null;
  stage: string;
  overall_score: number | null;
  summary: string | null;
  recommendation: string | null;
  criterion_scores: CriterionScore[];
  created_at: string;
  /** Populated client-side by joining interview data. */
  talk_ratio?: number | null;
  /** Populated client-side by joining interview data. */
  recording_url?: string | null;
}

export interface CoverageCriterion {
  criterion_name: string;
  status: AssessmentStatus;
  best_score: number | null;
  max_score: number;
  confidence_level: ConfidenceLevel | null;
  stages: string[];
}

export interface CoverageReport {
  assessed_count: number;
  total_required: number;
  coverage_ratio: number;
  criteria: CoverageCriterion[];
}

File: agentic-systems/talent-lens/backend/src/talentlens/schemas/assessment.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class EvidenceResponse(BaseModel):
    quote: str
    speaker: str | None
    relevance: str | None

    model_config = {"from_attributes": True}


class CriterionScoreResponse(BaseModel):
    criterion_name: str
    score: int
    max_score: int
    confidence_level: str = "demonstrated"
    assessment_status: str = "assessed_positive"
    reasoning: str | None
    evidence: list[EvidenceResponse] = []

    model_config = {"from_attributes": True}


class AssessmentResponse(BaseModel):
    id: uuid.UUID
    candidate_id: uuid.UUID
    interview_id: uuid.UUID | None
    stage: str
    overall_score: float | None
    summary: str | None
    recommendation: str | None
    criterion_scores: list[CriterionScoreResponse] = []
    created_at: datetime

    model_config = {"from_attributes": True}
